overlord/devicestate/devicestate.go

```diff
@@ -520,11 +520,23 @@ func Remodel(st *state.State, new *asserts.Model) ([]*state.TaskSet, error) {
 	// Set the new model assertion - this *must* be the last thing done
 	// by the change.
 	setModel := st.NewTask("set-model", i18n.G("Set new model assertion"))
-	setModel.Set("new-model", asserts.Encode(new))
 	for _, tsPrev := range tss {
 		setModel.WaitAll(tsPrev)
 	}
 	tss = append(tss, state.NewTaskSet(setModel))
 
-	return tss, nil
+	var msg string
+	if current.BrandID() == new.BrandID() && current.Model() == new.Model() {
+		msg = fmt.Sprintf(i18n.G("Refresh model assertion from revision %v to %v"), current.Revision(), new.Revision())
+	} else {
+		// TODO: add test once we support this kind of remodel
+		msg = fmt.Sprintf(i18n.G("Remodel device to %v/%v (%v)"), current.BrandID(), new.Model(), new.Revision())
+	}
+	chg := st.NewChange("remodel", msg)
+	for _, ts := range tss {
+		chg.AddAll(ts)
+	}
+	chg.Set("new-model", asserts.Encode(new))
```

Comment:
I would put it before the for loop, but that's me

Reply:
should we test for this explicitly or just implicitly? I do think we do the latter

Reply:
Also I forgot but we want string()  here,  assertion encode to utf8,   JSON encodes bytes to base64 which is a bit annoying for debugging and also ends up taking more space.  

Also with my incoming work this will become  remodCtx.Init(chg), shouldn't block this branch but FYI.

Reply:
Yeah, this was only tested implicitly, I added an explicit test now.